extensions/common/context.cc

```diff
@@ -391,6 +395,19 @@ void populateTCPRequestInfo(bool outbound, RequestInfo* request_info,
   request_info->request_protocol = kProtocolTCP;
 }
 
+bool populateGRPCInfo(RequestInfo* request_info) {
```

Comment:
1. This should also be called on a ticker.
2. Is there a way to get the message type? If the underlying protocol does not support it yet, we should add it there too.

Reply:
1. Will be in a followup per PR description. 
2. That's not possible in gRPC. We'd need the descriptor.

Reply:
I think if we can efficiently do proto when descriptor is known, it would be a big value add.

Reply:
With a descriptor, the message type is already known, so there is nothing extra being added. I think operation name would be useful, but it's not in the standard metrics.

Reply:
operation name for gRPC would likely be ok. it isn't in standard metrics because of unbounded cardinality, but within gRPC, that shouldn't be an issue.

Reply:
It should be easy to add with customization `operation_name: request.path` for gRPC metrics.